core/src/main/java/hudson/FilePath.java

```diff
@@ -2656,6 +2704,7 @@ private void readFromTar(String name, File baseDir, InputStream in) throws IOExc
                             _chmod(f, mode);
                     }
                 }
+                consumer.accept(f.getName());
```

Comment:
I think this is wrong - can you add a unit test for it?

if the tar entry is `foo/bar/wibble.txt` I think this will just pass `wibble.txt` to the consumer and not `foo/bar/wibble.txt`

Reply:
```suggestion
                consumer.accept(e.getName());
```

will give you the relative path in the tar, or 

```java
                consumer.accept(f.toPath().relativise(baseDir.toPath()).toString());
```

may well give you the relative path (untested)